### PR TITLE
NMS-9 op creation and ref implementation and CPU plugin

### PR DESCRIPTION
**Pull request by chenhu-wang**
### Details:
 - *NMS-9 op creation*
 - *NMS-9 op ref implementation and test*
 - *NMS-9 CPU plugin implementation*

### Tickets:
 - *79202*
 - *79203*
 - *79206*

### CODE REVIEW

**Review by tomdol — COMMENTED** [review]
Overall looks good, I've left some comments/questions. Also I think it would make sense to rename some variables like arg0, arg1, ..., func and similar to something more meaningful.

**Comment by tomdol** [comment]
Regarding the failing CI checks - it looks like the i64 data type support should resolve those fails.

File: src/core/src/op/non_max_suppression.cpp [hunk 1]
```diff
@@ -836,3 +836,350 @@ EnumNames<ngraph::op::v5::NonMaxSuppression::BoxEncodingType>::get() {
...
+
+bool ngraph::op::v9::NonMaxSuppression::visit_attributes(AttributeVisitor& visitor) {
+    NGRAPH_OP_SCOPE(v9_NonMaxSuppression_visit_attributes);
+    visitor.on_attribute("box_encoding", m_box_encoding);
+    visitor.on_attribute("sort_result_descending", m_sort_result_descending);
+    visitor.on_attribute("output_type", m_output_type);
+    return true;
+}
+
+void op::v9::NonMaxSuppression::validate_and_infer_types() {
```
[the remark this answers is not in the document]
Comment (chenhu-wang) [reply src/core/src/op/non_max_suppression.cpp:1107]: done, thanks!

File: src/core/reference/src/runtime/reference/non_max_suppression.cpp [hunk 2]
```diff
@@ -102,6 +102,141 @@ struct BoxInfo {
...
+
+            for (const auto& box_info : selected) {
+                filteredBoxes.push_back(box_info);
+            }
+        }
+    }
+
+    if (sort_result_descending) {
+        std::sort(filteredBoxes.begin(), filteredBoxes.end(), [](const BoxInfo& l, const BoxInfo& r) {
+            return (l.score > r.score) || (l.score == r.score && l.batch_index < r.batch_index) ||
```
[the remark this answers is not in the document]
Comment (chenhu-wang) [reply src/core/reference/src/runtime/reference/non_max_suppression.cpp]: done, thanks!

**Comment by chenhu-wang** [comment]
> Regarding the failing CI checks - it looks like the i64 data type support should resolve those fails.

fixed in cpu plugin, thanks!

**Comment by chenhu-wang** [comment]
> Overall looks good, I've left some comments/questions. Also I think it would make sense to rename some variables like arg0, arg1, ..., func and similar to something more meaningful.

renamed to name that is more meaningful. thanks!

**Comment by chenhu-wang** [comment]
@tomdol ,comments applied, could you please help take a look?

**Pull request merged**